spr/src/commands/diff.rs

```diff
@@ -41,12 +42,54 @@ pub struct DiffOptions {
     #[clap(long, short = 'm')]
     message: Option<String>,
 
+    /// Which commits in the branch should be created/updated. This can be a
+    /// revspec such as HEAD~4..HEAD~1 or just one commit like HEAD~7.
+    #[clap(long, short = 'r')]
+    refs: Option<String>,
+
     /// Submit this commit as if it was cherry-picked on master. Do not base it
     /// on any intermediate changes between the master branch and this commit.
     #[clap(long)]
     cherry_pick: bool,
 }
 
+fn get_oids(refs: &String, repo: &git2::Repository) -> Result<HashSet<Oid>> {
```

Comment:
```suggestion
fn get_oids(refs: &str, repo: &git2::Repository) -> Result<HashSet<Oid>> {
```
Functions should always take `&str` rather than `&String`. (Equivalent to taking `std::string_view` instead of `std::string const&` in C++).